apollo-router-core/src/query_planner/router_bridge_query_planner.rs

```diff
@@ -91,6 +96,33 @@ mod tests {
         insta::assert_debug_snapshot!(result);
     }
 
+    #[test]
+    fn empty_query_plan() {
+        let expected = PlannerResult::QueryPlan { node: None };
+        let actual: PlannerResult = serde_json::from_value(json!({ "kind": "QueryPlan"})).expect(
+            "If this test fails, It probably means QueryPlan::node isn't an Option anymore.\n
+                 Introspection queries return an empty QueryPlan, so the node field needs to remain optional.",
+        );
+
+        assert_eq!(expected, actual);
+    }
+
+    #[test(tokio::test)]
+    async fn empty_query_plan_should_be_a_planner_error() {
+        let expected = "Empty query plan. This often means an unhandled Introspection query was sent. Please file an issue to apollographql/router.";
+        let actual = RouterBridgeQueryPlanner::new(Arc::new(
+            include_str!("testdata/schema.graphql").parse().unwrap(),
+        ))
+        .get(
+            include_str!("testdata/unknown_introspection_query.graphql").into(),
+            None,
+            QueryPlanOptions::default(),
+        )
+        .await;
+
+        assert_eq!(expected, actual.unwrap_err().to_string());
```

Comment:
:yarn: Use insta, much easier
:yarn: Use `Default::default()` so we don't have to maintain the name of the actual struct.

```suggestion
        let result = RouterBridgeQueryPlanner::new(Arc::new(
            include_str!("testdata/schema.graphql").parse().unwrap(),
        ))
        .get(
            include_str!("testdata/unknown_introspection_query.graphql").into(),
            None,
            Default::default(),
        )
        .await;

        insta::assert_debug_snapshot!(result);
```